TAScheduler/UnitTests/tests.py

```diff
@@ -76,71 +77,879 @@ class TestAdminCourseInstrAsgmt(TestCase):  # Kevin
+    def test_duplicateCourse(self):
+        self.taObj.assignTACourse(self.courseDBList[0])
+        with self.assertRaises(ValueError,
+                               msg="violated integrity of database, can't assign a TA the same course twice"):
+            self.taObj.assignTACourse(self.courseDBList[0])
+
+    # 4] Adding Course to TA @ maxcap
+    def test_OverCap(self):
+        self.taObj.assignTACourse(self.courseDBList[0])
+        self.taObj.assignTACourse(self.courseDBList[1])
+        with self.assertRaises(ValueError,
+                               msg="can't assign courses when already reaches the max TA assignments"):
+            self.taObj.assignTACourse(self.courseDBList[2])
+
+    # 5] Trying to add a non-course
+    def test_NonCourse(self):
+        invalid_inputs = [123, 3.14, True, [1, 2, 3], {'key': 'value'}]  # testing a bunch of different obj types
+
+        for invalid_input in invalid_inputs:
+            with self.subTest(
+                    invalid_input=invalid_input):  # if 1 subtest test runs, it will continue running through loop
+                with self.assertRaises(ValueError, msg="Shouldn't be allowed to assign TA to non-course"):
+                    self.taObj.assignTACourse(invalid_input)
 
 
 class TestTAGetTACrseAsgmts(TestCase):  # Kiran
-    pass
+    taDB = None
+    courseDB = None
+    user = None  # for TA
+    taObj = None
+    adminObj = None  # for deleting course
+
+    def setUp(self):
+        self.user = User.objects.create(
+            email_address='TA@example.com',
+            password='TApassword',
+            first_name='TA',
+            last_name='User',
+            home_address='123 TA St',
+            phone_number='1234567890'
+        )
+        self.taDB = TA.objects.create(user=self.user, max_assignments=1)  # max 1 assignment!
+        self.taObj = TAObj(self.taDB)  # creating TA object using TA in database.
+        tempUserForAdmin = User(email_address="admin@example.com")
+        tempAdmin = Administrator(user=tempUserForAdmin)
+        self.adminObj = AdminObj(tempAdmin)
+
+        self.courseDB = Course.objects.create(
+            course_id=101,
+            semester='Fall 2023',
+            name='Introduction to Testing',
+            description='A course about writing tests in Django.',
+            num_of_sections=3,
+            modality='Online',
+            credits=4
+        )
+
+    # 1] 1 assignment
+    def test_1Assignment(self):
+        self.taObj.assignTACourse(self.courseDB)
+        self.assertEquals(self.taObj.getTACrseAsgmts(), 1, msg="should be 1 assigment")
+
+    # 2] 0 assignment
+    def test_0Assignment(self):
+        self.assertEquals(self.taObj.getTACrseAsgmts(), 0, msg="should be 0 assigments")
+
+    # 3] 1 "assignment" - not in db
+    def test_1AssignmentNoExistCourse(self):
+        tempCourse = Course(course_id=102)  # not "101", which exists already
+        self.taObj.assignTACourse(tempCourse)
```

Comment:
Again using methods from views_methods.py that you are not directly testing is setting you for errors. It feels like bad practice. Use the database built in stuff because that will work

Reply:
Also that method should throw a type error when you try to do this so this test would always fail